packages/client-node/src/connection/node_base_connection.ts

```diff
@@ -368,20 +369,23 @@ export abstract class NodeBaseConnection
     params: RunExecParams,
   ): Promise<ConnExecResult<Stream.Readable>> {
     const query_id = this.getQueryId(params.query_id)
-    const searchParams = toSearchParams({
+    const sendQueryInParams = params.values !== undefined
+    const toSearchParamsOptions = {
+      query: sendQueryInParams ? params.query : undefined,
       database: this.params.database,
       clickhouse_settings: params.clickhouse_settings,
       query_params: params.query_params,
       session_id: params.session_id,
       query_id,
-    })
+    }
+    const searchParams = toSearchParams(toSearchParamsOptions)
     const { controller, controllerCleanup } = this.getAbortController(params)
     try {
       const { stream, summary, response_headers } = await this.request(
         {
           method: 'POST',
           url: transformUrl({ url: this.params.url, searchParams }),
-          body: params.query,
+          body: sendQueryInParams ? params.values : params.query,
```

Comment:
nit: I'd prefer to keep a single logic path to send a request, but let's postpone until the next major release when we can *always* send `query` in URL and `values` as body (even if it's empty)